File: swt/src/std.r/rnd.f
      INTEGER LWB,UPB,NUM,I,SEED,TD(10)
      INTEGER GCTOI
      INTEGER ARG(128)
      INTEGER DASHL(3)
      INTEGER DASHU(3)
      INTEGER DASHN(3)
      INTEGER S1(2)
      INTEGER S100(4)
      DATA DASHL/173,236,0/
      DATA DASHU/173,245,0/
      DATA DASHN/173,238,0/
      DATA S1/177,0/
      DATA S100/177,176,176,0/
      CALL GETKWD(DASHL,ARG,128,S1)
      I=1
      LWB=GCTOI(ARG,I,10)
      CALL GETKWD(DASHU,ARG,128,S100)
      I=1
      UPB=GCTOI(ARG,I,10)
      CALL GETKWD(DASHN,ARG,128,S1)
      I=1
      NUM=GCTOI(ARG,I,10)
      CALL TIMDAT(TD,10)
      SEED=0
      DO 10000 I=1,10
        SEED=SEED+TD(I)
10000 CONTINUE
10001 CALL RND(IABS(SEED))
      I=1
      GOTO 10004
10002 I=I+1
10004 IF((I.GT.NUM))GOTO 10003
        CALL PRINT(-11,'*i*n.',INTS((UPB-LWB)*RND(0)+LWB))
      GOTO 10002
10003 CALL SWT
      END
C ---- Long Name Map ----

File: swt/src/std.r/rnd.r
# rnd --- generate random numbers

   integer lwb, upb, num, i, seed, td (10)
   integer gctoi

   character arg (MAXARG)

   string dash_l "-l"
   string dash_u "-u"
   string dash_n "-n"
   string s1 "1"
   string s100 "100"


   call getkwd (dash_l, arg, MAXARG, s1)     # get lower bound
   i = 1
   lwb = gctoi (arg, i, 10)

   call getkwd (dash_u, arg, MAXARG, s100)   # get upper bound
   i = 1
   upb = gctoi (arg, i, 10)

   call getkwd (dash_n, arg, MAXARG, s1)     # get number of values
   i = 1
   num = gctoi (arg, i, 10)

   call timdat (td, 10)
   seed = 0
   do i = 1, 10
      seed = seed + td (i)

   call rnd (iabs (seed))
   for (i = 1; i <= num; i = i + 1)
      call print (STDOUT, "*i*n"p, ints ((upb - lwb) * rnd (0) + lwb))

   stop
   end
